### PR TITLE
vimPlugins.{spacevim,SpaceVim}: point to spacevim; spacevim: 1.8.0 -> 2.3.0

### PR DESCRIPTION
**Pull request by PerchunPak**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by khaneliman** [comment]
Looks like we're getting eval failure on it

**Comment by PerchunPak** [comment]
Yeah, I was going to resolve it, but then broke my nixpkgs clone... multiple times...

Anyway, fixed now, thanks for a heads up!

**Comment by khaneliman** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 359304`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

**Comment by khaneliman** [comment]
Kinda confused on the two aliases atm, sorry if it's just because I'm not very familiar with them. It seems like this was a top level package so it makes sense we'd have a top level alias. What happens with the vim alias?

**Review by mrcjkb — CHANGES_REQUESTED** [review]
issue: These are two different things:

- The top-level package is [SpaceVim](https://github.com/SpaceVim/SpaceVim), a Vim/Neovim **distribution** (that is still maintained), **not a plugin**.
- `vimPlugins.spacevim` is an unmaintained (but not archived) plugin.

I would suggest to leave the top-level `SpaceVim` package as is and to deprecate or remove the unmaintained vim plugin.

**Comment by PerchunPak** [comment]
> Kinda confused on the two aliases atm, sorry if it's just because I'm not very familiar with them. It seems like this was a top level package so it makes sense we'd have a top level alias. What happens with the vim alias?

There are three of almost the same packages in nixpkgs

image


> The top-level package is [SpaceVim](https://github.com/SpaceVim/SpaceVim), a Vim/Neovim distribution (that is still maintained), not a plugin.

The top-level package hasn't been updated for almost four years (https://github.com/NixOS/nixpkgs/commit/1af5f62c1c00fee90fb461c991f1a1491e8aa19e), I would rather point users to auto-updatable version in vimPlugins. Also, we have multiple (Neo)Vim distributions in vimPlugins, so I don't see this as a problem. Note that top-level `spacevim` and `vimPlugins.SpaceVim` point to the same repository.

**Comment by khaneliman** [comment]
> > Kinda confused on the two aliases atm, sorry if it's just because I'm not very familiar with them. It seems like this was a top level package so it makes sense we'd have a top level alias. What happens with the vim alias?
> 
> There are three almost the same packages in nixpkgs

Thanks, didn't notice the vimPlugins.spacevim one at first.

**Comment by mrcjkb** [comment]
> Also, we have multiple (Neo)Vim distributions in vimPlugins, so I don't see this as a problem.

I do see this as a problem, for all distributions. (Neo)Vim distributions are not plugins.
Keeping distributions in a set of plugins is bad UX.

> I would rather point users to auto-updatable version

I don't think this is a good approach. With that reasoning, we could move neovim itself to `vimPlugins`.
Instead, we should ensure the packages can be [updated by the r-ryantm bot](https://github.com/NixOS/nixpkgs/pulls/r-ryantm).
Or, we could move them to a `vimDistributions` attrset and extend the updater script to also update the sources there.

> Note that top-level spacevim and vimPlugins.SpaceVim point to the same repository.

Does `vimPlugins.SpaceVim` actually work?
I can only find the generated package, but nothing in the overrides. The top-level `SpaceVim` package has quite a bit of build logic that I can't find anywhere for the `vimPlugins` version.
The [quick start guide](https://spacevim.org/quick-start-guide/) indicates that there is some build logic needed to get it working.

> The top-level package hasn't been updated for almost four years

If this is the case, and if `vimPlugins.SpaceVim` indeed does not work,  that would be a strong indicator that no Nix users are actually using SpaceVim anymore.
I would then suggest to ping the package maintainer and remove both packages.

**Comment by khaneliman** [comment]
> > Also, we have multiple (Neo)Vim distributions in vimPlugins, so I don't see this as a problem.
> 
> I do see this as a problem, for all distributions. (Neo)Vim distributions are not plugins. Keeping distributions in a set of plugins is bad UX.
> 

I know a couple distros that are moving towards a plugin based distribution to leverage the way users actually manage their configs instead of the overhead and complexity of doing it outside your plugin manager. 

> > I would rather point users to auto-updatable version
> 
> I don't think this is a good approach. With that reasoning, we could move neovim itself to `vimPlugins`. Instead, we should ensure the packages can be [updated by the r-ryantm bot](https://github.com/NixOS/nixpkgs/pulls/r-ryantm). Or, we could move them to a `vimDistributions` attrset and extend the updater script to also update the sources there.
> 

If a distribution is actually an involved process to get up and running, maybe.. but they're more than likely just a curation of plugins and configurations with some plugins for QOL things. 

> > Note that top-level spacevim and vimPlugins.SpaceVim point to the same repository.
> 
> Does `vimPlugins.SpaceVim` actually work? I can only find the generated package, but nothing in the overrides. The top-level `SpaceVim` package has quite a bit of build logic that I can't find anywhere for the `vimPlugins` version. The [quick start guide](https://spacevim.org/quick-start-guide/) indicates that there is some build logic needed to get it working.
> 

I haven't tried using it, but there are overrides added to fix check failures in https://github.com/NixOS/nixpkgs/pull/359307/files#diff-f1dfa85814fd1fb6e8b26a9c4b4f28585ba1020401c12600e17e781a931daadbR2427-R2437 I had never heard of it, so I don't know how popular it is.. 

> > The top-level package hasn't been updated for almost four years
> 
> If this is the case, and if `vimPlugins.SpaceVim` indeed does not work, that would be a strong indicator that no Nix users are actually using SpaceVim anymore. I would then suggest to ping the package maintainer and remove both packages.

I'm not sure I see the benefit in just removing instead of redirecting to use the vim plugin that is being updated.

**Comment by mrcjkb** [comment]
> I know a couple distros that are moving towards a plugin based distribution to leverage the way users actually manage their configs instead of the overhead and complexity of doing it outside your plugin manager.

Can we confirm that this is also the case for SpaceVim? Looking at the install script and the GitHub repo, it looks to me like this is not the case here.
Someone should test it by comparing the nix package and a non-nix SpaceVim install.

> but there are overrides added to fix check failures in

The `nvimRequireCheck` can help detect missing Lua dependencies of a Lua package, but it's not enough to confirm that a package (which is not unlikely to be more than just a Lua package) is working.
For example SpaceVim has quite a few non-standard directories that you won't find in `:h runtimepath`.
I'm assuming there's a reason for those to exist.

> I'm not sure I see the benefit in just removing instead of redirecting to use the vim plugin that is being updated.

My concern is that we appear to be redirecting to a "`vimPlugin`" that doesn't work as expected.

**Comment by PerchunPak** [comment]
> The [quick start guide](https://spacevim.org/quick-start-guide/) indicates that there is some build logic needed to get it working.

They just move the distribution in a folder where nvim can see it, and do some pretty logging. Nothing special in that script.

> Does `vimPlugins.SpaceVim` actually work?

Yeah, you are right. SpaceVim needs to be moved to ~/.config/nvim, or you have to edit init.toml in their repo somehow...

We probably should move distributions somewhere else from `vimPlugins`, but I don't care so much about these packages

CCing maintainer @fzakaria to hear their opinion

**Comment by fzakaria** [comment]
I haven't used SpaceVim in a long time; feel free to do what is right :)

**Comment by mrcjkb** [comment]
I can't seem to find the PR that added `vimPlugins.SpaceVim` either.
Let's remove them both. I think it's unlikely anyone is using it.

**Comment by PerchunPak** [comment]
Note that spacevim downloads its plugins itself impurely, but that's an issue to fix for someone else.

File: pkgs/applications/editors/vim/plugins/aliases.nix [hunk 1]
```diff
@@ -45,6 +45,8 @@ mapAliases (with prev; {
   coffeeScript        = vim-coffee-script; # backwards compat, added 2014-10-18
   Solarized           = vim-colors-solarized;
   solarized           = vim-colors-solarized;
+  spacevim            = throw "this distribution was unmaintained for the last 6 years, please use top-level 'spacevim'"; # added 2024-11-27
+  SpaceVim            = throw "this distribution didn't work properly in vimPlugins, please use top-level 'spacevim' instead"; # added 2024-11-27
```
Comment (khaneliman) [inline pkgs/applications/editors/vim/plugins/aliases.nix:49]: Assuming you actually tried using it and that's why you updated the top level one?

Comment (PerchunPak) [reply pkgs/applications/editors/vim/plugins/aliases.nix:49]: Yes, top-level one now works

image

**Review by teto — APPROVED** [review]

**Review by khaneliman — APPROVED** [review]

**Pull request merged**